mergebot_poller.py

```diff
@@ -1,47 +1,126 @@
-"""TODO(jasonkuster): DO NOT SUBMIT without one-line documentation for mergebot_poller.
-
-TODO(jasonkuster): DO NOT SUBMIT without a detailed description of mergebot_poller.
+"""
+mergebot_poller is a utility which polls an SCM and follows commands.
 """
 
-import mergebot
+from github_helper import GithubHelper
+import merge
 import Queue
+from threading import Thread
 
-GITHUB_SECRET = '../../github_auth/apache-beam.secret'
+# TODO: Fetch authorized users from somewhere official.
+AUTHORIZED_USERS = ["davorbonaci"]
+BOT_NAME = 'merge-bot'
+ORG = 'apache'
 
 class MergebotPoller:
-  scm_type = ""
-  repository = ""
-  branch = ""
-  scm_pollers = { "github": poll_github }
-  scm_mergers = { "github": mergebot.merge_git }
-  work_queue = Queue.Queue()
+  """
+  MergebotPoller is a class which handles the polling of a configured SCM.
+  """
+  SCM_POLLERS = { 'github': poll_github }
+  SCM_MERGERS = { 'github': merge.merge_git }
+  CMDS = { 'merge': CMD_MERGE }
+  CMD_MERGE = { 'github': merge_git }
+
+  def __init__(self, config):
+    self.config = config
 
+  def poll(self):
+    """
+    Kicks off polling of the SCM. Reads SCM type from config and starts the
+    appropriate poller.
 
-  def poll(self, config):
-    print('Starting merge poller for {}.'.format(config['name']))
-    self.scm_type = config['scm_type']
-    self.repository = config['repository']
-    self.branch = config['default_branch']
-    scm_pollers[scm_type](config)
+    :return: returns nothing
+    """
+    print('Starting merge poller for {}.'.format(self.config['name']))
```

Comment:
Some (all) of these prints should go through a Python logging library. I think those will do helpful things like allow us to set levels, include thread IDs, etc.